flink-table-store-core/src/main/java/org/apache/flink/table/store/table/sink/WriteBuilder.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.table.sink;
+
+import org.apache.flink.table.store.annotation.Experimental;
+import org.apache.flink.table.store.data.InternalRow;
+import org.apache.flink.table.store.types.RowType;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An interface for building the {@link TableWrite} and {@link TableCommit}.
+ *
+ * <p>Example of distributed batch writing:
+ *
+ * <pre>{@code
+ * // 1. Create a WriteBuilder (Serializable)
+ * Table table = catalog.getTable(...);
+ * WriteBuilder builder = table.newWriteBuilder();
+ *
+ * // 2. Write records in distributed tasks
+ * TableWrite write = builder.newWrite();
+ * write.write(...);
+ * write.write(...);
+ * write.write(...);
+ * List<CommitMessage> messages = write.prepareCommit(true, 0);
+ *
+ * // 3. Collect all CommitMessages to a global node
+ *
+ * // 3.1 commit
+ * TableCommit commit = builder.newCommit();
+ * if (commit.filterCommitted(0)) {
+ *     // It has already been committed. In order to avoid
+ *     // repeated committing, return directly.
+ *     return;
+ * }
+ * commit.commit(0, allCommitMessages());
+ *
+ * // 3.2. Expire snapshots and partitions
+ * commit.expireSnapshots();
+ * commit.expirePartitions();
+ *
+ * }</pre>
+ *
+ * @since 0.4.0
+ */
+@Experimental
+public interface WriteBuilder extends Serializable {
+
+    /** A name to identify this table. */
+    String tableName();
+
+    /** Returns the row type of this table. */
+    RowType rowType();
+
+    /** Get commit user, set by {@link #withCommitUser}. */
+    String commitUser();
+
+    /**
+     * Set commit user, the default value is UUID. The commit user used by {@link TableWrite} and
+     * {@link TableCommit} must be the same, otherwise there will be some conflicts.
+     */
+    WriteBuilder withCommitUser(String commitUser);
```

Comment:
> Set commit user, the default value is UUID.

the default value is a random UUID.

> he commit user used by TableWrite and TableCommit must be the same, otherwise there will be some conflicts.

Users cannot set commitUser separately for `TableWrite` and `TableCommit` with public API. No need for this statement.

Reply:
> Users cannot set commitUser separately for TableWrite and TableCommit with public API. No need for this statement.

User may use different `WriteBuilder`, I think we can add this setence.